### PR TITLE
[Chore](build) enable -Wextra and remove some -Wno

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
# Proposed changes

1. enable -Wextra and remove some -Wno

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
3. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
5. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
6. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/CMakeLists.txt [hunk 1]
```diff
@@ -441,12 +441,13 @@ check_function_exists(sched_getcpu HAVE_SCHED_GETCPU)
 
 # compiler flags that are common across debug/release builds
 #  -Wall: Enable all warnings.
-#  -Wno-sign-compare: suppress warnings for comparison between signed and unsigned
 #    integers
 #  -pthread: enable multithreaded malloc
 #  -DBOOST_DATE_TIME_POSIX_TIME_STD_CONFIG: enable nanosecond precision for boost
 #  -fno-omit-frame-pointers: Keep frame pointer for functions in register
-set(CXX_COMMON_FLAGS "${CXX_COMMON_FLAGS} -Wall -Wno-sign-compare -pthread -Werror")
+set(CXX_COMMON_FLAGS "${CXX_COMMON_FLAGS} -Wall -Wextra -pthread -Werror")
```
Comment (adonis0147) [inline be/CMakeLists.txt:448]:
```suggestion
set(CXX_COMMON_FLAGS "${CXX_COMMON_FLAGS} -Wall -Wextra -pthread -Werror -pedantic")
```

Comment (adonis0147) [reply be/CMakeLists.txt:448]: Can add an extra option.

Comment (BiteTheDDDDt) [reply be/CMakeLists.txt:448]: I tried pedantic, still many compilation warnings, maybe these can be done in other pr after

Comment (adonis0147) [reply be/CMakeLists.txt:448]: > I tried pedantic, still many compilation warnings, maybe these can be done in other pr after

Okey, it is a great job.

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 36.64 seconds
 load time: 496 seconds
 storage size: 17123165845 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230112105109_clickbench_pr_78547.html

File: be/src/http/action/pprof_actions.cpp [hunk 2]
```diff
@@ -228,9 +228,11 @@ void CmdlineAction::handle(HttpRequest* req) {
...
     }
     char buf[1024];
-    // Ignore unused return value
-    if (fscanf(fp, "%1023s ", buf))
-        ;
+
+    if (fscanf(fp, "%1023s ", buf) != 1) {
+        return;
+    }
+
```
Comment (adonis0147) [inline be/src/http/action/pprof_actions.cpp:235]: Maybe we should call `fclose` here, or use RAII to manage the `FILE` pointer.

File: be/src/util/bfd_parser.cpp [hunk 3]
```diff
@@ -102,9 +102,11 @@ BfdParser* BfdParser::create() {
...
 
     char prog_name[1024];
-    // Ignore unused return value
-    if (fscanf(file, "%1023s ", prog_name))
-        ;
+
+    if (fscanf(file, "%1023s ", prog_name) != 1) {
+        return nullptr;
+    }
+
```
Comment (adonis0147) [inline be/src/util/bfd_parser.cpp:109]: Ditto.

File: be/src/util/pprof_utils.cpp [hunk 4]
```diff
@@ -66,9 +66,10 @@ Status PprofUtils::get_self_cmdline(std::string* cmd) {
         return Status::InternalError("Unable to open file: /proc/self/cmdline");
     }
     char buf[1024];
-    // Ignore unused return value
-    if (fscanf(fp, "%1023s ", buf))
-        ;
+
+    if (fscanf(fp, "%1023s ", buf) != 1) {
+        return Status::InternalError("get_self_cmdline read buffer failed");
+    }
```
Comment (adonis0147) [inline be/src/util/pprof_utils.cpp:72]: Ditto.

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**